api/v1alpha1/microvmcluster_types.go

```diff
@@ -28,6 +28,24 @@ type MicrovmClusterSpec struct {
 	// MicrovmProxy is the proxy server details to use when calling the microvm service. This is an
 	// alteranative to using the http proxy environment variables and applied purely to the grpc service.
 	MicrovmProxy *Proxy `json:"microvmProxy,omitempty"`
+	// TLSSecretRef is a reference to the name of a secret which contains TLS cert information
+	// for connecting to Flintlock hosts. It is recommended that each flintlock host is configured
+	// with its own cert signed by a common CA, and set to use mTLS.
+	// The CAPMVM client should be provided with the CA, and a client cert and key signed by that CA.
+	// The secret should be created in the same namespace as the MicroVMCluster.
+	// The secret should contain the following data:
+	//
+	// apiVersion: v1
+	// kind: Secret
+	// metadata:
+	// 	name: mytlssecret
+	// type: Opaque
```

Comment:
I'm thinking that we should use secrets of type `kubernetes.io/tls`? Which would then make the keys:

- tls.crt
- tls.key
- ca.crt

What do you think?

Reply:
https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets

Reply:
oh yeh! I have that in my head at the start, and then it fell out 😆

Reply:
seems to be that the tls.key and tls.cert are commonly stored in a tls/secret, and the ca cert in a configmap or regular opaque secret 🤔

Reply:
i think because if you create the secret via the cli, you only have `--cert` and `--key`:

```
kubectl create secret tls server --cert X --key Y
```

which become the tls.cert and tls.key keys. and it wont know what to do with anything else.

If you create it via yaml, you can have whatever keys you like, so we could tell people to do that. but then they would not be able to create it via the cli which would suck.

so we are going to have 2 secrets. one for tls and one for the ca

Reply:
Thats interesting, we can still have 3 in the same secret....cert-manager does this. But yeah i get the point that it would be a pain in the arse from the cli.